modules/core/src/main/java/org/apache/synapse/endpoints/EndpointDefinition.java

```diff
@@ -208,7 +216,18 @@ public String getAddress(MessageContext messageContext) {
      * @param address the absolute address to be used
      */
     public void setAddress(String address) {
-        this.address = address;
+        this.originalAddress = address;
+        String fetchedAddress = ConfigUtils.fetchEnvironmentVariables(address);
```

Comment:
We don't need fetchedAddress local variable. We can directly assign it to this.address.

Reply:
Resolved https://github.com/apache/synapse/pull/55/commits/80314352fca33dba35454bd2aa6af3402fa8b151